coffeescript-loader/loader.js

```diff
@@ -25,41 +34,50 @@ export function resolve(specifier, context, defaultResolve) {
   return defaultResolve(specifier, context, defaultResolve);
 }
 
-export function getFormat(url, context, defaultGetFormat) {
+export async function load(url, context, defaultLoad) {
   // Now that we patched resolve to let CoffeeScript URLs through, we need to
   // tell Node.js what format such URLs should be interpreted as. Because
   // CoffeeScript transpiles into JavaScript, it should be one of the two
   // JavaScript formats: 'commonjs' or 'module'.
   if (extensionsRegex.test(url)) {
-    // CoffeeScript files can be either CommonJS or ES modules, so we want any
-    // CoffeeScript file to be treated by Node.js the same as a .js file would
-    // be at the same location (based on the "type" field in the nearest
-    // parent package.json file). To determine how Node.js would interpret an
-    // arbitrary .js file, append .js to our CoffeeScript URL and see what
-    // format Node.js returns for such a URL. If 'commonjs' is returned, a
-    // handler will need to be registered with require.extensions to process
-    // that file via the CommonJS loader.
-    const { format } = defaultGetFormat(`${url}.js`);
-    return {format};
-  }
+    // CoffeeScript files can be either CommonJS or ES modules, but since load
+    // handles both format and source based on the same url, it cannot be used
+    // for non-js files. Instead, a quick search up the filesystem for a
+    // package.json with a `type` field settles the format issue.
+    const format = await getPackageType(dirname(fileURLToPath(url)));
 
-  // Let Node.js handle all other URLs.
-  return defaultGetFormat(url, context, defaultGetFormat);
-}
+    // source is ignored (never checked) for cjs, so safe to omit
+    if (format === 'commonjs') return { format };
```

Comment:
What does your comment here mean? We’re also losing important information in the original comment. Perhaps something like this? And we also always include braces for `if` blocks, I think.

```suggestion
    // If this hook returns a format of 'commonjs', the source will be ignored.
    // To handle CommonJS files, a handler will need to be registered with
    // require.extensions in order to process the files with the CommonJS loader.
    // Avoiding the need for a separate CommonJS handler is a future enhancement
    // planned for ES module loaders.
    if (format === 'commonjs') {
      return { format };
    }
```

Reply:
🤔 I've seen many instances of the curly brackets omitted in source, but a quick look thru the docs suggests superfluous curly brackets are present. If you realllllllly want.